protocol/Cargo.toml

```diff
@@ -10,15 +10,18 @@ rust-version = "1.63.0"
 
 [features]
 default = ["std"]
+# High-level wrappers using futures traits.
 async = ["std", "futures/std"]
+# High-level wrappers using tokio traits - may affect MSRV requirements.
 tokio = ["std", "tokio/io-util"]
```

Comment:
I''m confused how is this not rejected for not using `dep:`.

Reply:
`dep:` is new to me, and we were originally targeting MSRV 1.56 here so technically wasn't available (it was introduced in 1.60). [Digging into it though](https://doc.rust-lang.org/cargo/reference/features.html#optional-dependencies), it sounds like `dep` is a way to make things a little more explicit. I believe things are just working right now because this tokio feature flag is overriding the implicit one created by the optional tokio dependency? In any case, would rather be more explicit on intentions.

I pushed 5443dac4ce138cde82883a5654d6b92ebb20c8a8 to switch to the dep syntax.

Reply:
Well, that's the thing. IIRC Cargo complained before when hitting same-named things that they are conflicting. That's what I'm confused about - why it wasn't rejected by it here. But yes, `dep:` is more explicit and thus nicer.

Reply:
I can't for the life of me find any docs on how explicit features interact with the implicit ones, but it hasn't been exploding so I think it must overwrite or merge them somehow.

Reply:
I don't know it by heart either but I remember I had to do a bit more digging when I already knew it's possible but didn't know the details. Someone slacked on "how do we teach this" section in the RFC :D